src/BiDirectionalLinkedList.h

```diff
@@ -0,0 +1,50 @@
+//
+// Created by shvmo on 02/05/2020.
+//
+
+#ifndef DATASTRUCTURESWET1_BIDIRECTIONALLINKEDLIST_H
+#define DATASTRUCTURESWET1_BIDIRECTIONALLINKEDLIST_H
+
+
+#include "BiDirectionalNode.h"
+#include <iostream>
+
+using std::ostream;
+
+template<class Type>
+class BiDirectionalLinkedList {
+public:
+    BiDirectionalLinkedList();
+
+    ~BiDirectionalLinkedList();
+
+    shared_ptr<BiDirectionalNode<Type>> getFirst() const;
+
+    // Done for optimization when we want to iterate a bit faster
+    BiDirectionalNode<Type> *getFirstPointer() const;
+
+    shared_ptr<BiDirectionalNode<Type>> getLast() const;
+
+    void insert(const Type &newValue);
+
+    shared_ptr<BiDirectionalNode<Type>> findNodeByValue(const Type &valueToFind) const;
+
+    void deleteNodeByValue(const Type &valueToFind);
```

Comment:
if we use the term `remove()` then we should use it here too instead of `delete`

Reply:
fixxxxxxxxxed